src/test/helpers/e2e-test-utils.ts

```diff
@@ -0,0 +1,796 @@
+import * as assert from "assert";
+import * as vscode from "vscode";
+import { CardData, SymbolCardData } from "../../EventTypes";
+import { LocationFinder } from "../../location-finder";
+import { TEST_CARDS } from "../fixtures/card-data";
+import { MockMiroClient } from "../mocks/mock-miro-client";
+import { waitFor } from "../suite/test-utils";
+import { TestPortManager } from "./test-port-manager";
+
+/**
+ * Type guard to check if a card is a symbol card
+ */
+export function isSymbolCard(card: CardData): card is SymbolCardData {
+  return card.type === "symbol";
+}
+
+/**
+ * E2E Test utilities for MockMiroClient testing
+ */
+export class E2ETestUtils {
+  private static mockClient: MockMiroClient | null = null;
+  private static locationFinder: LocationFinder = new LocationFinder();
+  private static testMiroServer: any = null; // MiroServer instance for testing
+  private static cardStorage: any = null; // CardStorage instance
+  private static serverCardStorage: any = null; // ServerCardStorage instance
+  private static capturedEvents: Map<string, any[]> = new Map(); // Event capture for testing
+
+  /**
+   * Initialize test port allocation for the test suite
+   * This should be called once at the beginning of the test suite
+   */
+  static async initializeTestPort(): Promise<number> {
+    const port = await TestPortManager.allocateTestPort();
+    console.log(`[E2ETestUtils] Initialized test port: ${port}`);
+
+    // Check if production port is in use
+    const productionInUse = await TestPortManager.isProductionPortInUse();
+    if (productionInUse) {
+      console.log(
+        "[E2ETestUtils] Production AppExplorer instance detected on port 9042 - using test port to avoid conflicts",
+      );
+    }
+
+    return port;
```

Comment:
This can be simplified by always using a non-production test port to avoid conflicts

```suggestion
    return  TestPortManager.isProductionPortInUse();
```